### ISSUE TITLE
Add ability to have a custom url for Jenkins Master used by Mesos plugins. (Use custom url and not getRootURL)

### ISSUE THREAD

**Issue by tangicolin** [jenkinsci/mesos-plugin#82]
Hello,
Mesos plugins use getRootURL here :https://github.com/jenkinsci/mesos-plugin/blob/master/src/main/java/org/jenkinsci/plugins/mesos/MesosCloud.java#L167
If I understand well, it's needed by mesos to send back offers to Mesos Jenkins Master.  

In my setup, i have jenkins and mesos "on cloud" and i use some haproxy frontend for routing http request to jenkins Master. I want to avoid that traffic between mesos and jenkins pass thought my haproxy frontend. So for that i can set IP:PORT in jenkins url. But since 1.572, a reverse proxy check is performed by jenkins and warning is generated if jenkins url is wrong . See 
https://wiki.jenkins-ci.org/display/JENKINS/Jenkins+says+my+reverse+proxy+setup+is+broken

So i will be a nice feature if we can put a custom url into mesos jenkins plugin configuration. (And have getRootURL only as fallback if this custom url is not specified)

**Comment by tomstockton** [comment]
+1 - I have exactly the same use case as @tangicolin

**Comment by cornelf** [comment]
:+1:

**Comment by maselvaraj** [comment]
@tangicolin @tomstockton @cornelf  I also need this as much as you do :). I tried to implement this long time back but since I faced some issues and I had to concentrate on other stuff, I did not continue later. 

It is not straight forward as it seems. Even if we place a custom URL in Mesos plugin to  download slave.jar/JNLP file.  Slave.jar would still try to connect to the root url configured in Jenkins. 

Anyways, I will work on this, will keep you posted!

**Comment by maselvaraj** [comment]
@tangicolin  @tomstockton @cornelf https://github.com/jenkinsci/mesos-plugin/pull/92 - addresses this issue.

It would be great if you can build the plugin, test it and let me know how it goes.

**Comment by maselvaraj** [comment]
This is implemented in v0.7.0

**Comment by scottcarey** [comment]
This no longer seems to work.   
 With mesos cloud on the newest version (after update) everything breaks. 
It uses the cloud specified Jenkins URL  (call it internal-host-name) to pull the agent.jar, but then uses the global Jenkins URL (call it external-jenkins-hame) when the agent tries to register.

The logs do this successfully on a mesos node:

```
I1222 05:47:42.944906 14158 fetcher.cpp:444] Fetching URI 'http://internal-host-name:8080/jnlpJars/agent.jar'
```

Then fail later with:
```
SEVERE: https://external-jenkins-name/ provided port:46481 is not reachable
```

The former name is configured in the mesos cloud "Jenkins URL" but the latter is the global one used by the webapp.
